File: assignments/clojure/binary/binary_test.clj
(ns binary.test (:require [clojure.test :refer :all]))
(load-file "binary.clj")

(deftest binary-1-is-decimal-1
  (is (= 1 (binary/to-decimal "1"))))

(deftest binary-10-is-decimal-2
  (is (= 2 (binary/to-decimal "10"))))

(deftest binary-11-is-decimal-3
  (is (= 3 (binary/to-decimal "11"))))

(deftest binary-100-is-decimal-4
  (is (= 4 (binary/to-decimal "100"))))

(deftest binary-1001-is-decimal-9
  (is (= 9 (binary/to-decimal "1001"))))

(deftest binary-11010-is-decimal-26
  (is (= 26 (binary/to-decimal "11010"))))

(deftest binary-10001101000-is-decimal-1128
  (is (= 1128 (binary/to-decimal "10001101000"))))

(deftest invalid-binary-is-decimal-0
  (is (= 0 (binary/to-decimal "carrot"))))

(run-tests)

File: assignments/elixir/binary/binary_test.exs
if System.get_env("EXERCISM_TEST_EXAMPLES") do
  Code.load_file("example.exs")
else
  Code.load_file("binary.exs")
end

ExUnit.start

defmodule BinaryTest do
  use ExUnit.Case, async: true

  test "binary 1 is decimal 1" do
    assert 1 == Binary.to_decimal("1")
  end

  test "binary 10 is decimal 2" do
    assert 2 == Binary.to_decimal("10")
  end

  test "binary 11 is decimal 3" do
    assert 3 == Binary.to_decimal("11")
  end

  test "binary 100 is decimal 4" do
    assert 4 == Binary.to_decimal("100")
  end

  test "binary 1001 is decimal 9" do
    assert 9 == Binary.to_decimal("1001")
  end

  test "binary 11010 is decimal 26" do
    assert 26 == Binary.to_decimal("11010")
  end

  test "binary 10001101000 is decimal 1128" do
    assert 1128 == Binary.to_decimal("10001101000")
  end

  test "invalid binary is decimal 0" do
    assert 0 == Binary.to_decimal("carrot")
  end
end
